compiler/query_structure.rs

```diff
@@ -193,18 +193,12 @@ impl ParametrisedPipelineStructure {
         PipelineStructure { parametrised_structure: self, parameters, variable_names, available_variables }
     }
 
-    pub fn always_taken_blocks(&self) -> Vec<QueryStructureBlockID> {
+    pub fn must_have_been_satisfied_conjunctions(&self) -> Vec<QueryStructureConjunctionID> {
         self.stages
             .iter()
             .filter_map(|stage| match stage {
-                QueryStructureStage::Match { block: QueryStructureConjunction { conjunction } } => match conjunction.first() {
-                    Some(QueryStructurePattern::Block { index }) => Some(index),
-                    Some(_) | None => {
-                        debug_assert!(!conjunction.iter().any(|c| matches!(c, QueryStructurePattern::Block { .. })));
-                        None
-                    }
-                },
-                QueryStructureStage::Insert { block }
+                QueryStructureStage::Match { block }
```

Comment:
This is the only exception to renaming block -> conjunction, because we call them blocks in TypeDB.
We could call them conjunction too, but it's the "root" conjunction.